wingstructure/aero/liftingline.py

```diff
@@ -74,30 +81,23 @@ def __init__(self):
         self._airbrake = None
         self._aoa = None
         self._control_surfaces = {}
-    
-    def calculate(self, C_L, controls:dict={}, airbrake:bool=False, airfoil_db:dict=None, options:dict=None):
-        
-        c_ls = np.zeros_like(self.ys)
-        C_Di = 0.0
 
-        # collect contributions to lift distribution
-        contributions = [self._base]
-        if airbrake: contributions.append(self._airbrake)
-        
-        contributions += [self._calc_controlsurface(*cs) for cs in controls.items()]
+        self._area = 0.0
+        self._spanwidth = 0.0
+    
+    def calculate(self, C_L, controls:dict={}, airbrake:bool=False, all_results:bool=False):
         
-        for contrib in contributions:
-            C_Di += contrib.C_Di
-            c_ls += contrib.c_ls
-            C_L -= contrib.C_L
+        α, res = self.__call__(C_L, 'C_L', controls, airbrake)
 
-        # choose angle of attack to reach C_L
-        α = C_L / self._aoa.C_L
+        if all_results:
+            # Moment coefficient in flight direction
+            A = self._area
+            b = self._spanwidth
+            C_Mx = np.trapz(self.ys*res.c_ls*self.chords, self.ys)/ (A*b)
 
-        c_ls += α * self._aoa.c_ls
-        C_Di += abs(α) * self._aoa.C_Di
+            return np.rad2deg(α), res.c_ls, res.C_Di, C_Mx
 
-        return np.rad2deg(α), c_ls, C_Di
+        return np.rad2deg(α), res.c_ls
```

Comment:
Maybe we should keep the api consistent and return None for fields that are not calculated.

Reply:
Valid point, but I think this behavior is not uncommon in python.